### DISCUSSION TITLE
Failing to mount using PlanetScale

### DISCUSSION THREAD

**Discussion by lewisl9029** [juicedata/juicefs#2479] [General]
**What happened**:

I tried to set up a quick test using PlanetScale and B2 and managed to get past the format step, but got stuck when trying to mount, with the following error:

> 2022/08/01 04:13:12.173360 juicefs[32408] <FATAL>: new session: create session: update table session2, delslices: Error 1105: symbol `COLUMNS`.ORDINAL_POSITION not found [mount.go:439]

**What you expected to happen**:

Mount is successful

**How to reproduce it (as minimally and precisely as possible)**:

1. Create a new PlanetScale db (without promoting any branches to production as that will lock the schema)
2. Create a new Backblaze B2 bucket (though I believe the issue should be reproducible using any storage provider since error sounds like a DB error)
3. Run `juice format` using PlanetScale connection information for default branch (using format `mysql://username:pw@tcp(hostname)/dbname?tls=true`)
4. Run `juice mount` using same information

Expected:
Mounts successfully

Actual:
Fails with following error:
> 2022/08/01 04:13:12.173360 juicefs[32408] <FATAL>: new session: create session: update table session2, delslices: Error 1105: symbol `COLUMNS`.ORDINAL_POSITION not found [mount.go:439]

**Anything else we need to know?**

Also saw the following warnings above:

```
2022/08/01 04:13:11.481558 juicefs[32408] <WARNING>: The latency to database is too high: 257.687ms [sql.go:199]
2022/08/01 04:13:11.547956 juicefs[32408] <WARNING>: the database does not support read-only transaction [sql.go:673]
```

Could the lack of read-only transaction be an issue here?

Let me know if there's any other information I can provide to help debug. Thanks!

**Environment**:
- JuiceFS version (use `juicefs --version`) or Hadoop Java SDK version:
- Cloud provider or hardware configuration running JuiceFS: juicefs version 1.0.0-rc3+2022-07-14.d1fe59d 
- OS (e.g `cat /etc/os-release`): Windows 11
- Kernel (e.g. `uname -a`): N/A
- Object storage (cloud provider and region, or self maintained): Backblaze B2 us-west-001
- Metadata engine info (version, cloud provider managed or self maintained): PlanetScale 
- Network connectivity (JuiceFS to metadata engine, JuiceFS to object storage): tested on a wired 1gbps up/down connection
- Others:

**Comment by davies** [comment]
It seems that PlanetScale is not compatible with MySQL, please file the issue to PlanetScale?

**Comment by davies** [comment]
JuiceFS requires transaction to ensure the atomics and integration of meta operations, PlanetScale may not work well even we fix the compatibility issues.
